job/strategies/semantic_similarity.py

```diff
@@ -1,131 +1,88 @@
 import logging
-import time
-from typing import Any, Dict, List
+from typing import Dict, List, Optional
 
 import numpy as np
 import pandas as pd
 from sentence_transformers import SentenceTransformer
 from sklearn.preprocessing import normalize
 
-from db.helpers import create_model, refresh_db, set_current_model
 from db.mappings.model import ModelType
-from db.mappings.recommendation import Rec
-from job.helpers.itertools import batch
-from job.helpers.knn import KNN, map_neighbors_to_recommendations
 from job.strategies.templates.strategy import Strategy
 from lib.config import config
 from sites.templates.site import Site
 
-# TODO: When adding functions to job.py in the future, maybe move these job.py and pass to respective functions as a parameter?
-PRETRAINED_MODEL_NAME = config.get("SS_ENCODER")
-MAX_RECS = config.get("MAX_RECS")
+# TODO: When adding functions to job.py in the future, maybe move these job.py and pass to respective functions as a
+#  parameter?
 
 
 class SemanticSimilarity(Strategy):
     """
     Semantic-Similarity site configs and methods.
     """
 
-    def fetch_data(self, site: Site, interactions_data: pd.DataFrame) -> List[Dict[str, Any]]:
+    def __init__(self):
+        super().__init__(model_type=ModelType.SEMANTIC_SIMILARITY)
+
+        # Other attributes
+        self.article_data: Optional[List[Dict[str, str]]] = None
+        self.interactions_data: Optional[pd.DataFrame] = None
+        self.PRETRAINED_MODEL_NAME = config.get("SS_ENCODER")
+
+    def fetch_data(self, site: Site, interactions_data: pd.DataFrame = None):
         """
         Fetch data of all articles included in the interactions table.
         """
+        self.site = site
```

Comment:
⛏ I'd turn this single line into a method of `Strategy`, e.g., `Strategy.set_site`. It doesn't have a whole lot to do with the logic of fetching additional data, so best to separate it to reduce coupling.